File: ts/student.ts
import {Degree} from "./degree"

export default class Student {
    private id: string = ""
    private First_Name: string = ""
    private Last_Name: string = ""
    private Email: string = ""
    private Age: number = 0
    private Student_Degree: Degree|null = null
    private CourseDays: Array<Number> = []

    constructor(id:string, firstname:string, lastname:string, email:string, age:number, degree:Degree, coursedays:Array<Number>) {
        this.setID(id)
        this.setFirstName(firstname)
        this.setLastName(lastname)
        this.setEmail(email)
        this.setAge(age)
        this.setDegree(degree)
        this.setCourseDays(coursedays)
    }

    getID():string {return this.id}
    getFirstName():string {return this.First_Name}
    getLastName():string {return this.Last_Name}
    getEmail():string {return this.Email}
    getAge():Number {return this.Age}
    getDegree():Degree|null {return this.Student_Degree}
    getCourseDays():Array<Number> {return this.CourseDays}

    setID(id:string):void {this.id = id}
    setFirstName(name:string):void {this.First_Name = name}
    setLastName(name:string):void {this.Last_Name = name}
    setEmail(email:string):void {this.Email = email}
    setAge(age:number):void {this.Age = age}
    setDegree(degree: Degree):void {this.Student_Degree = degree}
    setCourseDays(days:Array<Number>):void {this.CourseDays = days}


    Print():void {
        console.log(`${this.getID()}    ${this.getFirstName()}    ${this.getLastName()}    ${this.getEmail()}    ${this.getEmail()}    ${this.getAge()}    ${this.getDegree()}    ${this.getCourseDays()}`)
    }
}

File: cpp/cpp/student.cpp
#include "student.h"
#include "degree.h"
#include <iostream>


using namespace std;

Student::Student(string id, string firstname, string lastname, string email, int age, int coursedays[], Degree degree) {
	setID(id);
	setFirstName(firstname);
	setLastName(lastname);
	setEmail(email);
	setAge(age);
	setDegree(degree);
	setCourseDays(coursedays);
};

Student::~Student() {};

//Getters
string Student::getID() { return ID; }

string Student::getFirstName() { return First_Name; }

string Student::getLastName() { return Last_Name; }

string Student::getEmail() { return Email; }

int Student::getAge() { return Age; }

Degree Student::getDegree() { return Student_Degree; }

int* Student::getCourseDays() { return CourseDays; }

//Setters
void Student::setID(string id) { ID = id; return;}

void Student::setFirstName(string name) { First_Name = name; return;}

void Student::setLastName(string name) { Last_Name = name; return;}

void Student::setEmail(string email) { Email = email; return;}

void Student::setAge(int age) { Age = age; return;}

void Student::setDegree(Degree degree) { Student_Degree = degree; return;}

void Student::setCourseDays(int coursedays[3]) {

	for (int i = 0; i < 3; i++) {
		CourseDays[i] = coursedays[i];
	}

	return; 
}

void Student::Print() {
	cout << getID() << "\t\t";
	cout << getFirstName() << "\t\t";
	cout << getLastName() << "       \t";
	cout << getEmail() << "     \t";
	cout << getAge() << "\t\t";
	cout << "{" << getCourseDays()[0] << "," << getCourseDays()[1] << "," << getCourseDays()[2] << "}" << "\t\t";
	cout << getDegreeName(getDegree()) << endl;

	return;
}